oauthproxy.go

```diff
@@ -820,32 +820,52 @@ func (p *OAuthProxy) addHeadersForProxying(rw http.ResponseWriter, req *http.Req
 		req.Header["X-Forwarded-User"] = []string{session.User}
 		if session.Email != "" {
 			req.Header["X-Forwarded-Email"] = []string{session.Email}
+		} else {
+			req.Header.Del("X-Forwarded-Email")
 		}
 	}
+
 	if p.PassUserHeaders {
 		req.Header["X-Forwarded-User"] = []string{session.User}
 		if session.Email != "" {
 			req.Header["X-Forwarded-Email"] = []string{session.Email}
+		} else {
+			req.Header.Del("X-Forwarded-Email")
 		}
 	}
+
 	if p.SetXAuthRequest {
 		rw.Header().Set("X-Auth-Request-User", session.User)
 		if session.Email != "" {
 			rw.Header().Set("X-Auth-Request-Email", session.Email)
-		}
-		if p.PassAccessToken && session.AccessToken != "" {
-			rw.Header().Set("X-Auth-Request-Access-Token", session.AccessToken)
+		} else {
+			rw.Header().Del("X-Auth-Request-Email")
 		}
 	}
-	if p.PassAccessToken && session.AccessToken != "" {
-		req.Header["X-Forwarded-Access-Token"] = []string{session.AccessToken}
+
+	if p.PassAccessToken {
+		if session.AccessToken != "" {
```

Comment:
It looks like you have split the if conditions by mistake. Breaking the if statement in to two leaves a branch un handled.

Reply:
This way makes some sense because it always applies if you enable the `--pass-access-token` option. This option is something you have complete control over and does not vary with client state. If you don't enable this option, presumably your other applications do not expect this header to be set by `oauth2_proxy`, and you may prefer for oauth2_proxy to not mess with the parts of requests that do concern it. You might not be aware of all the headers that oauth2_proxy has options to set, and might be confused by some headers which you intended to pass through not being passed.

I'd also be fine with oauth2_proxy clearing all headers it could possibly be configured to set, not just those which its current configuration may set. There's not _too_ many optionally-enabled headers yet. I'm just explaining the reasoning for the way it was done here so far.

Reply:
The split is deliberate, for the reasons @ploxiln outlines - however setting different header *keys* in the next few lines is a copy/paste error. I'll do another pass.